src/arcanum.rs

```diff
@@ -0,0 +1,93 @@
+use crate::util::Result;
+use html2md::parse_html;
+use reqwest::Url;
+use serde::Deserialize;
+use serenity::builder::CreateEmbed;
+
+pub struct Arcanum {
+    base_url: String,
+    icon: String,
+}
+
+#[derive(Debug, Deserialize)]
+pub struct Entry {
+    id: u64,
+    event: u64,
+    event_name: String,
+    event_date: String,
+    date: String,
+    paraphrased: bool,
+    modified_date: String,
+    tags: Vec<String>,
+    lines: Vec<Line>,
+    note: String,
+}
+
+#[derive(Debug, Deserialize)]
+pub struct PaginatedEntries {
+    pub count: u64,
+    next: Option<String>,
+    previous: Option<String>,
+    pub results: Vec<Entry>,
+}
+
+#[derive(Debug, Deserialize)]
+pub struct Line {
+    speaker: String,
+    text: String,
+}
+
+impl Arcanum {
+    async fn raw_request(&self, endpoint: &str, params: &Vec<(&str, &str)>) -> Result<String> {
```

Comment:
```suggestion
    async fn raw_request(&self, endpoint: &str, params: &[(&str, &str)]) -> Result<String> {
```
`&Vec<T>` isn't really any more useful than `&[T]` here.